### DISCUSSION TITLE
How can I set `wait_for_completion` dynamically while using `TriggerDagRunOperator`?

### DISCUSSION THREAD

**Discussion by rogue-gamer-ryt** [apache/airflow#31117] [Q&A] [answered]
I want to control `wait_for_completion` parameter from my dag_run.conf. I tried the below code and it didn't work as expected. Would really appreciate if there is a way to achieve it,

```python
    trigger_dag_2 = TriggerDagRunOperator(
        task_id="trigger_trigger_dag_2,
        trigger_dag_id="dag_2",
        wait_for_completion=str_to_bool("{{ dag_run.conf.get('wait_for_completion')}}"),
        poke_interval=10,
    )
```

The `str_to_bool` checks if it is a bool value then it direclty returns the bool value, if the value is like `'true'`/`'True'` (string value) it would convert it bool `True` and by default it returns `False`. I am using this (`str_to_bool `) function inside a `python_callable` function where is works fine, over here it is not working as expected.

**Comment by rogue-gamer-ryt** [comment, marked as answer]
Okay so I think why it is not working properly is because only certain arguments for an operator can use template values. For `TriggerDagRunOperator` it is `template_fields: Sequence[str] = ("trigger_dag_id", "trigger_run_id", "execution_date", "conf")` so I am not able to set `wait_for_completion` correctly. 

If this is the reason can we add `wait_for_completion` to template_fields` ?

**Reply by potiuk** [reply]
Yes. Feel free to add PR to add it to template_fields

**Reply by potiuk** [reply]
Also - for now - if you want to use it befor it gets merged and released, you are free to create your own custom operator by extending the original one add making the field template_field there.

**Reply by rogue-gamer-ryt** [reply]
Sure, that works - I have raised a [PR](https://github.com/apache/airflow/pull/31122) based on your comments, can I add you as a reviewer?

**Reply by potiuk** [reply]
Sure - some static checks/formatting need to be fixe but it's close.

**Reply by rogue-gamer-ryt** [reply]
Pushed the fixes.
